File: src/biaobiaoqi/algorithm/oj/pat/advancedlevel/APAT1048.java
package biaobiaoqi.algorithm.oj.pat.advancedlevel;

import java.util.Scanner;
/**
 * It will timeout using Java.
 * The working code is in APAT1048.cpp.
 * @author biaobiaoqi
 *
 */
public class APAT1048 {
	public static int max = 100005;
	public static int[] a = new int[max];
	public static void main(String[] atgs){
		Scanner cin = new Scanner(System.in);
		int n = cin.nextInt();
		int m = cin.nextInt();
		int min = max;
		int tmpMin, tmp;
		for (int i =0; i != n; i ++) {
			tmp = cin.nextInt();
			if (tmp < m) {
				tmpMin = tmp < m - tmp? tmp : m - tmp;
				if (a[m - tmp] != 0 && tmpMin < min) {
					min = tmpMin;
				}
				a[tmp] = 1;				
			}
		}
		
		if (min < max) {
			System.out.println(min + " " + (m - min));
		}else {
			System.out.println("No Solution");
		}
	}
}

File: src/biaobiaoqi/algorithm/oj/pat/advancedlevel/APAT1048.cpp
//
//  main.c
//  BBQ
//
//  Created by biaobiaoqi  on 13-4-17.
//  Copyright (c) 2013年 biaobiaoqi. All rights reserved.
//

#include <stdio.h>
const int NUM = 100005;
int a[NUM];

int main(int argc, const char * argv[])
{
    int min = NUM;
    int n, m;
    int tmp;
    scanf("%d%d", &n, &m);
    for (int i = 0; i != n; ++i) {
        scanf("%d",&tmp);
        if (tmp < m) {
            int tmpMin = tmp < m - tmp ? tmp : m - tmp;
            if (a[m - tmp] == 1 && tmpMin < min) {
                min = tmpMin;
            }
            a[tmp] = 1;            
        }
    }
    
    if (min != NUM) printf("%d %d\n", min, m - min);
    else printf("No Solution");
    
    return 0;
}

